Configuration Cache error

## 🐛 Describe the bug
Error when running `dependencyGraph`

```
FAILURE: Build failed with an exception.

* What went wrong:
Configuration cache problems found in this build.

41 problems were found storing the configuration cache, 1 of which seems unique.
- Task `:dependencyGraph` of type `io.github.adityabhaskar.dependencygraph.plugin.DependencyGraphTask`: invocation of 'Task.project' at execution time is unsupported.
  See https://docs.gradle.org/8.2/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution
```

## ⚠️ Current behavior
Errors when running task but generates the files.

## ✅ Expected behavior
No error, support configuration cache correctly

## 💣 Steps to reproduce
Enable gradle configuration cache and run `dependencyGraph`

## 📷 Screenshots


## 📱 Tech info
 - Library/App version: 0.1.4

--- a/README.md
+++ b/README.md
@@ -24,14 +24,14 @@ The plugin is available in the [Gradle plugins repository](https://plugins.gradl
 Kotlin:
 ```kotlin
 plugins {
-  id("io.github.adityabhaskar.dependencygraph") version "0.1.4"
+  id("io.github.adityabhaskar.dependencygraph") version "<version>"
 }
 ```
 
 Groovy:
 ```groovy
 plugins {
-  id "io.github.adityabhaskar.dependencygraph" version "0.1.4"
+  id "io.github.adityabhaskar.dependencygraph" version "<version>"
 }
 ```
 
@@ -97,7 +97,7 @@ classDef andNode fill:#baffc9,color:#333333;
 classDef javaNode fill:#ffb3ba,color:#333333;
 
 %% Modules
-subgraph  
+subgraph
   direction LR;
   :example:feature{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/feature/dependencyGraph.md' style='text-decoration:auto'>:example:feature</a>}}:::javaNode;
   :example:models{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/models/dependencyGraph.md' style='text-decoration:auto'>:example:models</a>}}:::javaNode;
@@ -132,7 +132,7 @@ classDef andNode fill:#baffc9,color:#333333;
 classDef javaNode fill:#ffb3ba,color:#333333;
 
 %% Modules
-subgraph  
+subgraph
   direction LR;
   :example:data{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/data/dependencyGraph.md' style='text-decoration:auto'>:example:data</a>}}:::javaNode;
   :example:domain[<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/domain/dependencyGraph.md' style='text-decoration:auto'>:example:domain</a>]:::javaNode;

--- a/README.md
+++ b/README.md
@@ -97,7 +97,7 @@ classDef andNode fill:#baffc9,color:#333333;
 classDef javaNode fill:#ffb3ba,color:#333333;
 
 %% Modules
-subgraph
+subgraph  
   direction LR;
   :example:feature{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/feature/dependencyGraph.md' style='text-decoration:auto'>:example:feature</a>}}:::javaNode;
   :example:models{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/models/dependencyGraph.md' style='text-decoration:auto'>:example:models</a>}}:::javaNode;
@@ -132,7 +132,7 @@ classDef andNode fill:#baffc9,color:#333333;
 classDef javaNode fill:#ffb3ba,color:#333333;
 
 %% Modules
-subgraph
+subgraph  
   direction LR;
   :example:data{{<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/data/dependencyGraph.md' style='text-decoration:auto'>:example:data</a>}}:::javaNode;
   :example:domain[<a href='https://github.com/adityabhaskar/Gradle-dependency-graphs/blob/main/example/domain/dependencyGraph.md' style='text-decoration:auto'>:example:domain</a>]:::javaNode;

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -4,6 +4,8 @@ kotlin = "1.9.23"
 ktlintGradle = "11.6.1"
 pluginPublish = "1.2.1"
 versionCheck = "0.51.0"
+android-lint = "8.3.0"
+android-lint-gradle = "1.0.0-alpha01"
 
 [plugins]
 detekt = { id = "io.gitlab.arturbosch.detekt", version.ref = "detekt"}
@@ -11,6 +13,8 @@ kotlin = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin"}
 ktlint = { id = "org.jlleitschuh.gradle.ktlint", version.ref = "ktlintGradle"}
 pluginPublish = { id = "com.gradle.plugin-publish", version.ref = "pluginPublish"}
 versionCheck = { id = "com.github.ben-manes.versions", version.ref = "versionCheck"}
+lint = { id = "com.android.lint", version.ref = "android-lint"}
 
 [libraries]
 junit = "junit:junit:4.13.2"
+android-lint-gradle = { module = "androidx.lint:lint-gradle", version.ref = "android-lint-gradle" }

--- a/plugin-build/gradle.properties
+++ b/plugin-build/gradle.properties
@@ -1,5 +1,5 @@
 ID=io.github.adityabhaskar.dependencygraph
-VERSION=0.1.5
+VERSION=0.1.6
 GROUP=io.github.adityabhaskar
 DISPLAY_NAME=Gradle module dependency graphs
 DESCRIPTION=A plugin to automatically produce Github/mermaid compatible dependency graphs

--- a/plugin-build/plugin/build.gradle.kts
+++ b/plugin-build/plugin/build.gradle.kts
@@ -1,15 +1,17 @@
 import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
 
-@Suppress("DSL_SCOPE_VIOLATION")
 plugins {
     kotlin("jvm")
     `java-gradle-plugin`
     alias(libs.plugins.pluginPublish)
+    alias(libs.plugins.lint)
 }
 
 dependencies {
     implementation(kotlin("stdlib"))
     implementation(gradleApi())
+
+    lintChecks(libs.android.lint.gradle)
 
     testImplementation(libs.junit)
 }

--- a/plugin-build/plugin/src/main/kotlin/io/github/adityabhaskar/dependencygraph/plugin/core/ParseGraph.kt
+++ b/plugin-build/plugin/src/main/kotlin/io/github/adityabhaskar/dependencygraph/plugin/core/ParseGraph.kt
@@ -65,9 +65,9 @@ internal fun parseDependencyGraph(
             javaProjects.add(project)
         }
 
-        project.configurations.all { config ->
+        project.configurations.configureEach { config ->
             config.dependencies
-                .filterIsInstance(ProjectDependency::class.java)
+                .filterIsInstance<ProjectDependency>()
                 .map { it.dependencyProject }
                 .filter { it.path !in ignoredModules }
                 .forEach { dependency ->
@@ -100,7 +100,7 @@ internal fun parseDependencyGraph(
         val allDependencies = it.configurations
             .map { config ->
                 config.dependencies
-                    .filterIsInstance(ProjectDependency::class.java)
+                    .filterIsInstance<ProjectDependency>()
                     .filter { dependency ->
                         dependency.dependencyProject.path != it.path
                     }

--- a/plugin-build/settings.gradle.kts
+++ b/plugin-build/settings.gradle.kts
@@ -2,6 +2,7 @@ pluginManagement {
     repositories {
         gradlePluginPortal()
         mavenCentral()
+        google()
     }
 }
 

